exp/api/v1beta1/machinepool_types.go

```diff
@@ -60,6 +60,13 @@ type MachinePoolSpec struct {
 	// FailureDomains is the list of failure domains this MachinePool should be attached to.
 	// +optional
 	FailureDomains []string `json:"failureDomains,omitempty"`
+
+	// Whether the Replicas value is externally managed. This is useful when
+	// the infrastructure is a managed machine pool, or if the cluster-autoscaler is scaling
+	// the underlying cloud infrastructure directly outside of cluster-api. Set this to true
+	// if you wish the replica count to be managed outside of cluster-api. Defaults to false.
+	// +optional
```

Comment:
Should we document this property and the expected behavior of infra providers in the book? https://cluster-api.sigs.k8s.io/developer/architecture/controllers/machine-pool.html#infrastructure-provider

Reply:
Yes, I will definitely update the book. And also yes, the logic you mention does not change.